### PR TITLE
GHA: Remove ref from checkout

### PR DESCRIPTION
**Pull request by janisz**
## Description

Fixes: https://github.com/stackrox/stackrox/pull/4127#discussion_r1053243794

### CODE REVIEW

**Comment by janisz** [comment]
Current dependencies on/for this PR:
* master
  * **PR #4213** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4213" target="_blank">Graphite</a>  👈
    * **PR #4127** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4127" target="_blank">Graphite</a> 
      * **PR #4143** <a href="https://app.graphite.dev/github/pr/stackrox/stackrox/4143" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/stackrox/stackrox/4213?utm_source=stack-comment).

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -22,9 +22,7 @@ jobs:
     steps:
       - name: Checkout
         uses: actions/checkout@v3
-        with:
-          fetch-depth: 0
```
Comment (msugakov) [inline .github/workflows/build.yaml:26]: Was there any value in `fetch-depth: 0`?

Comment (janisz) [reply .github/workflows/build.yaml:26]: you're right `make tag` will not work with shallow clone. We should fix this as there is no sense in pulling the whole history

**Comment by janisz** [comment]
Closing as it looks like this way of checkout is required to prow builds work

**Pull request closed without merging**